commit 277352b6cbeda5c0976e56d25e3fcd775db96305
Author: Eric Sandeen <sandeen@redhat.com>
Date:   Mon Mar 3 11:12:12 2025 -0600

    f2fs: consolidate unsupported option handling errors
    
    When certain build-time options are disabled, some mount options are not
    accepted. For quota and compression, all related options are dismissed
    with a single error message. For xattr, acl, and fault injection, each
    option is handled individually. In addition, inline_xattr_size was missed
    when CONFIG_F2FS_FS_XATTR was disabled.
    
    Collapse xattr, acl, and fault injection errors into a single string, for
    simplicity, and handle the missing inline_xattr_size case.
    
    Signed-off-by: Eric Sandeen <sandeen@redhat.com>
    Reviewed-by: Chao Yu <chao@kernel.org>
    Signed-off-by: Jaegeuk Kim <jaegeuk@kernel.org>

diff --git a/fs/f2fs/super.c b/fs/f2fs/super.c
index 40c78f526f94..1d48900d8cd3 100644
--- a/fs/f2fs/super.c
+++ b/fs/f2fs/super.c
@@ -775,16 +775,11 @@ static int parse_options(struct super_block *sb, char *options, bool is_remount)
 			break;
 #else
 		case Opt_user_xattr:
-			f2fs_info(sbi, "user_xattr options not supported");
-			break;
 		case Opt_nouser_xattr:
-			f2fs_info(sbi, "nouser_xattr options not supported");
-			break;
 		case Opt_inline_xattr:
-			f2fs_info(sbi, "inline_xattr options not supported");
-			break;
 		case Opt_noinline_xattr:
-			f2fs_info(sbi, "noinline_xattr options not supported");
+		case Opt_inline_xattr_size:
+			f2fs_info(sbi, "xattr options not supported");
 			break;
 #endif
 #ifdef CONFIG_F2FS_FS_POSIX_ACL
@@ -796,10 +791,8 @@ static int parse_options(struct super_block *sb, char *options, bool is_remount)
 			break;
 #else
 		case Opt_acl:
-			f2fs_info(sbi, "acl options not supported");
-			break;
 		case Opt_noacl:
-			f2fs_info(sbi, "noacl options not supported");
+			f2fs_info(sbi, "acl options not supported");
 			break;
 #endif
 		case Opt_active_logs:
@@ -922,11 +915,8 @@ static int parse_options(struct super_block *sb, char *options, bool is_remount)
 			break;
 #else
 		case Opt_fault_injection:
-			f2fs_info(sbi, "fault_injection options not supported");
-			break;
-
 		case Opt_fault_type:
-			f2fs_info(sbi, "fault_type options not supported");
+			f2fs_info(sbi, "fault injection options not supported");
 			break;
 #endif
 		case Opt_lazytime: